diff --git a/src/plugins/cnat/cnat_node_feature.c b/src/plugins/cnat/cnat_node_feature.c
index 0108d09dc..a82086fbf 100644
--- a/src/plugins/cnat/cnat_node_feature.c
+++ b/src/plugins/cnat/cnat_node_feature.c
@@ -43,6 +43,7 @@ cnat_input_feature_new_flow_inline (vlib_main_t *vm, vlib_buffer_t *b,
 				    cnat_timestamp_t *ts)
 {
   vlib_combined_counter_main_t *cntm = &cnat_translation_counters;
+  cnat_snat_policy_entry_t *cpe = cnat_snat_policy_entry_get_default ();
   const cnat_translation_t *ct = NULL;
   cnat_timestamp_rewrite_t *rw = NULL;
   cnat_client_t *cc;
@@ -54,6 +55,8 @@ cnat_input_feature_new_flow_inline (vlib_main_t *vm, vlib_buffer_t *b,
   udp_header_t *udp0;
   index_t cti;
 
+  u32 in_if = vnet_buffer (b)->sw_if_index[VLIB_RX];
+  int ispod;
   if (AF_IP4 == af)
     {
       ip4 = vlib_buffer_get_current (b);
@@ -112,7 +115,9 @@ cnat_input_feature_new_flow_inline (vlib_main_t *vm, vlib_buffer_t *b,
       rw->tuple.port[VLIB_TX];
 
   ts->trk = trk0;
-  if (trk0->ct_flags & CNAT_TRK_FLAG_NO_NAT)
+  ispod =
+    clib_bitmap_get (cpe->interface_maps[CNAT_SNAT_IF_MAP_INCLUDE_POD], in_if);
+  if (trk0->ct_flags & CNAT_TRK_FLAG_NO_NAT && !ispod)
     {
       const dpo_id_t *dpo0;
       const load_balance_t *lb1;
